backend/src/main/java/com/gaucho/sos/controllers/AuthenticationController.java

```diff
@@ -0,0 +1,55 @@
+package com.gaucho.sos.controllers;
+
+import com.gaucho.sos.domain.User;
+import com.gaucho.sos.dtos.LoginRequestDTO;
+import com.gaucho.sos.dtos.RegisterRequestDTO;
+import com.gaucho.sos.dtos.AuthResponseDTO;
+import com.gaucho.sos.repositories.UserRepository;
+import com.gaucho.sos.security.services.JwtService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/authentication")
+public class AuthenticationController {
+    @Autowired
+    private UserRepository repository;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+    @Autowired
+    JwtService jwtService;
+
+    @PostMapping("/login")
+    public ResponseEntity login( @RequestBody LoginRequestDTO body ) {
```

Comment:
Aqui no meu VSCode apareceu esse aviso:

<img width="904" alt="Captura de Tela 2024-10-02 às 19 51 22" src="https://github.com/user-attachments/assets/64161464-10d8-4166-b0a5-374326962f95">

Sugestão pra definir o tipo da `ResponseEntity`:
```suggestion
    public ResponseEntity<AuthResponseDTO> login( @RequestBody LoginRequestDTO body ) {
```

Reply:
Resolvido.